reconcile/terraform_resources.py

```diff
@@ -201,16 +201,20 @@ def setup(
     include_accounts: Optional[Collection[str]],
     exclude_accounts: Optional[Collection[str]],
 ) -> tuple[
-    ResourceInventory, Optional[OCMap], Terraform, ExternalResourceSpecInventory
+    ResourceInventory,
+    Optional[OCMap],
+    Terraform,
+    ExternalResourceSpecInventory,
+    Iterable[str],
 ]:
     accounts = queries.get_aws_accounts(terraform_state=True)
+    all_accounts = tuple(a["name"] for a in accounts)
```

Comment:
`account_names` is used in caller check in https://github.com/app-sre/qontract-reconcile/pull/3790/files#diff-0e9f20c0944ab8088807c726d964c2df24b5c33b873c4c4ef1a4a78c5fe86517R232-R234, bug or expected?

Reply:
I think you mean https://github.com/app-sre/qontract-reconcile/pull/3790#discussion_r1323817962

Reply:
not only that, but here https://github.com/app-sre/qontract-reconcile/pull/3790/files#diff-0e9f20c0944ab8088807c726d964c2df24b5c33b873c4c4ef1a4a78c5fe86517L124, not sure if it just want to filter current account names, or check all account names

Reply:
ah, I see. Then this pr would not actually work as expected. If the current resources do not match with the caller, they will never be added to ResourceInventory. https://github.com/BumbleFeng/qontract-reconcile/blob/234229993e1a433a2b27d40681e4f62dfdbfba86/reconcile/terraform_resources.py#L96-L99

Reply:
IIRC, the original idea was to reduce the current state items. Unlike openshift-saas-deploy deploys resources in a single namespace, tf-r applies secrets across different namespaces. the secrets may also come for other integrations. Filter secrets by the caller(account names) can keep the current state items as little as necessary.